src/main/java/org/codehaus/mojo/taglist/FileAnalyser.java

```diff
@@ -71,52 +69,52 @@
     /**
      * The character encoding of the files to analyze.
      */
-    private String encoding;
-    
+    private final String encoding;
+
     /**
      * The Locale of the files to analyze.
      */
-    private Locale locale;
+    private final Locale locale;
 
     /**
      * The directories to analyze.
      */
-    private Collection sourceDirs;
+    private final Collection<String> sourceDirs;
 
     /**
      * The files to include, as a comma separated list of patterns.
      */
-    private String includes;
+    private final String includes;
 
     /**
      * The files top exclude, as a comma separated list of patterns.
      */
-    private String excludes;
+    private final String excludes;
 
     /**
      * Log for debug output.
      */
-    private Log log;
+    private final Log log;
 
     /**
      * Set to true if the analyzer should look for multiple line comments.
      */
-    private boolean multipleLineCommentsOn;
+    private final boolean multipleLineCommentsOn;
 
     /**
      * Set to true if the analyzer should look for tags without comments.
      */
-    private boolean emptyCommentsOn;
+    private final boolean emptyCommentsOn;
 
     /**
      * String used to indicate that there is no comment after the tag.
      */
-    private String noCommentString;
+    private final String noCommentString;
 
     /**
      * ArrayList of tag classes.
      */
-    private List<TagClass> tagClasses = new ArrayList();
+    private final List<TagClass> tagClasses;
```

Comment:
not sure using final make any improvement except make lines longer

Reply:
Will do, I didn't understand  that the "extends:" statement references a local repo.
Will update the tags accordingly. 👍🏻

Reply:
Yes.
It will be much easier to spot PRs making a field non-final anymore. It shows intent.
It *might* help the compiler, but I think the compiler is already good enough.

Reply:
@olamy what about consistency?
Most write-once fields are `final` already.

Let's not nit-pick about this because the line gets 6 chars longer. :( 

![image](https://user-images.githubusercontent.com/1413391/150977735-c5bd4d1f-e96d-40f5-8d6e-642867ab7984.png)